app/store/charmstore-api.js

```diff
@@ -407,6 +407,53 @@ YUI.add('charmstore-api', function(Y) {
       var bundle = jsyaml.safeLoad(bundleYAML);
       var wrapped = { 'bundle-deploy': bundle };
       return jsyaml.safeDump(wrapped);
+    },
+
+    /**
+      Gets the list of available versions of the supplied charm id.
+
+      @method getAvailableVersions
+      @param {String} charmId The charm id to fetch all of the versions for.
+      @param {Function} successCallback The success callback.
+      @param {Function} failureCallback The failure callback.
+    */
+    getAvailableVersions: function(charmId, successCallback, failureCallback) {
+      charmId = charmId.replace('cs:', '');
+      var series = charmId.split('/')[0];
+      this._makeRequest(
+          this._generatePath(charmId, null, '/expand-id'),
```

Comment:
Unless I'm misunderstanding how this function is to be used it looks like you're not actually using the result of this function call, could that be right?